### DISCUSSION TITLE
Rancher deployment

### DISCUSSION THREAD

**Discussion by Compukid** [netbox-community/netbox-docker#390] [Getting Help] [answered]
I am new to docker / rancher. I have followed other tutorials and able to figure out how to deploy other docker images into rancher fine. for instance Plex, pihole, papermerge, heimdall and guacamole to name a few but I am having some trouble with the netbox image. My rancher deployment is as follows

- Docker image: netboxcommunity/netbox 
- configured container port 8080 and hostport as 8006 (Available)
- have 1 environment variable SECRET_KEY  
- setup a volume for /config directory

I didn't think I needed anything else, but I see in the log I get an error could not connect to server: address not available

> 	Is the server running on host "localhost" (127.0.0.1) and accepting 
	TCP/IP connections on port 5432? 
could not connect to server: Address not available 
	Is the server running on host "localhost" (::1) and accepting 
	TCP/IP connections on port 5432? 
 
❌ Waited 30s or more for the DB to become ready. 

The log keeps repeating that until 30 seconds goes by and then the pod restarts :-( 
I know I am missing something, but having some trouble figuring it out. any help / guidance is appreciated.

**Comment by cimnine** [comment, marked as answer]
You will need a PostgreSQL database and at least one (IMO better two) Redis servers. Currently you will also need an nginx container which mounts a volume from the Netbox container, but this will change in the upcoming release. You need to run a second instance of Netbox (the worker instance) to handle the webhooks and do other behind-the-scenes jobs.

In order to set it up yourself I suggest to take a hard look at our `docker-compose.yml` for the minimal requirements. Or check out [netbox-kubernetes](https://github.com/CENGN/netbox-kubernetes) for a complete and read-to-use Kubernetes deployment (Rancher is a Kubernetes distribution) of Netbox Docker.

**Reply by Compukid** [reply]
oh, thank you so much for the quick response. This makes more sense. I watched a video where a script installed "everything" on just a docker instance. I will do more research on the netbox kubernetes and hopefully I can figure it out. 

thanks again.

**Comment by eduncan911** [comment]
When using Kubernetes, you should be looking into Helm Chart deployments - not Docker Compose (though you still could, with the right compose.yaml).

Helm Charts is to Kubernetes what Docker-Compose is to Docker.  That is, it bundles all dependencies and configurations, and deploys it into pods and creates any statefulsets (volumes) needed.  It's wonderful!

Since you are on Rancher, you should already have Helm setup.  Read through Rancher's docs a little more to see how to use Helm Charts.  Then...

While I don't have a K8s deployment at hand to test, the community has already put together a Helm Chart for Netbox:

https://github.com/search?q=helm+netbox

IMO, stick to whichever is actively updated (e.g. bootc at the time of this writing).
